src/core/document/query.pl

```diff
@@ -141,13 +141,13 @@
         do_or_die(prefix_expand_schema(Query_Type, Prefixes, Query_Type_Ex),
                   error(query_error(unknown_prefix(Query_Type)),_)),
         do_or_die(once(class_subsumed(DB, Type, Query_Type_Ex)),
-                  error(query_error(not_a_a_subclass(Type, Query_Type_Ex)), _))
+                  error(query_error(not_a_subclass(Type, Query_Type_Ex)), _))
     ;   var(Type)
     ->  throw(error(query_error(unknown_type_for_query(Query)), _))
     ;   Query_Type_Ex = Type),
 
     do_or_die(type_descriptor(DB, Query_Type_Ex, Query_Type_Descriptor),
-              error(query_error(type_not_found(Query_Type)), _)),
+              error(query_error(type_not_found(Query_Type_Ex)), _)),
```

Comment:
I believe this changes `Blah` to `http://somewhere.for.now/schema#Blah` in the output of the test case below? If so, I think it's better to report the user input exactly, which is `Blah`. Otherwise, it adds extra cognitive load to try to match the error report with the input.

Reply:
I agree, but there is actually no unexpanded type in the context at the moment. The `Query_Type` is unbound. To get the unprefixed type would require additional machinery.